src/rules/noWarningCommentsRule.ts

```diff
@@ -0,0 +1,175 @@
+            Two arguments may be optionally provided:
+
+            * \`"location"\` specifies where to look for 'warning' terms in comments:
+                * 'start' specifies at the start.
+                * 'anywhere' specifies anywhere in the comment.
+            * \`"terms"\` specifies a list of terms (case-insensitive) to be considered 'warning' terms.
+
+            By default, \`"location"\` defaults to \`"start"\` and \`"terms"\` defaults to
+            \`["todo", "fixme", "xxx"]\`.
+            `,
+        options: {
+            type: "object",
+            additionalProperties: false,
+            properties: {
+                terms: {
+                    type: "array",
+                    items: {
+                        type: "string",
+                    },
+                },
+                location: {type: "string"},
+            },
+        },
+        optionExamples: [
+            true,
+            [true, {location: "anywhere"}],
+            [true, {location: "start", terms: ["todo", "fixme", "xxx"]}],
+        ],
+        type: "style",
+        typescriptOnly: false,
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static COMMENT_ERROR_FACTORY = (term: string, location: "start" | "anywhere"): string =>
+        `Warning term '${term}' found ${location === "start" ? "at start of" : "in"} comment`
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithFunction(sourceFile, walk, parseOptions(this.ruleArguments[0] as IOptionsObject));
+    }
+}
+
+function parseOptions(options: IOptionsObject): Options {
+    if (typeof options === "undefined" || options === null) {
+        options = {};
+    }
+
+    if (
+        (typeof options.location !== "undefined" && options.location !== null) &&
+        options.location !== "start" &&
+        options.location !== "anywhere") {
+        throw new Error(`Option 'location' value is invalid: ${options.location}`);
+    }
+
+    if (
+        (typeof options.terms !== "undefined" && options.terms !== null) &&
+        !(options.terms instanceof Array)) {
+        throw new Error("Option 'terms' must be a string array.");
+    }
+
+    return {
+        location: (typeof options.location !== "undefined" && options.location !== null) ?
+            options.location as "start" | "anywhere" :
+            "start",
+        terms: (typeof options.terms !== "undefined" && options.terms !== null) ?
+            options.terms :
+            ["todo", "fixme", "xxx"],
+    };
+}
+
+function checkCommentContainsWarningTerms(comment: string, regExps: RegExp[], rng: ts.CommentRange, ctx: Lint.WalkContext<Options>) {
+    regExps.forEach((regExp) => {
+        const match = regExp.exec(comment);
+        if (typeof match !== "undefined" && match !== null) {
+            ctx.addFailure(
+                rng.pos + match[1].length,
+                rng.pos + match[1].length + match[2].length,
```

Comment:
suggestion: if you use `ctx.addFailureAt` you only can pass the width of the error instead of the end

Reply:
Does it matter?  This works fine and is explicit.

Reply:
it's just a suggestion, feel free to ignore it